lib/features/send/send_did_page.dart

```diff
@@ -38,80 +58,145 @@ class SendDidPage extends HookWidget {
-                    controller: controller,
-                    onTapOutside: (event) => focusNode.unfocus(),
-                    maxLines: null,
-                    enableSuggestions: false,
-                    autocorrect: false,
-                    decoration: InputDecoration(
-                      labelText: Loc.of(context).didPrefix,
-                    ),
-                    validator: (value) {
-                      if (value == null ||
-                          !value.startsWith(Loc.of(context).didPrefix)) {
-                        return Loc.of(context).invalidDid;
-                      }
-                      return null;
-                    }),
+            ),
+            Expanded(
+              child: TextFormField(
+                focusNode: focusNode,
+                controller: controller,
+                onTap: () => errorText.value = null,
+                onTapOutside: (_) async {
+                  if (controller.text.isNotEmpty) {
+                    errorText.value = await _isValidDid(controller.text)
+                        ? null
+                        : errorMessage;
+                  }
+                  focusNode.unfocus();
+                },
+                maxLines: null,
+                enableSuggestions: false,
+                autocorrect: false,
+                decoration: InputDecoration(
+                    labelText: Loc.of(context).didPrefix,
+                    errorText: errorText.value),
+                validator: (value) => value == null
+                    ? Loc.of(context).thisFieldCannotBeEmpty
+                    : null,
               ),
-            ],
-          )),
+            ),
+          ],
+        ),
+      ),
     );
   }
 
   Widget _buildScanQrTile(
-      BuildContext context, TextEditingController controller) {
-    return ListTile(
-      leading: const Icon(Icons.qr_code),
-      title: Text(
-        Loc.of(context).scanQrCode,
-        style: Theme.of(context).textTheme.bodyMedium,
+    BuildContext context,
+    TextEditingController controller,
+    ValueNotifier<String?> errorText,
+    String did,
+  ) {
+    return Padding(
+      padding: const EdgeInsets.symmetric(vertical: Grid.xxs),
+      child: ListTile(
+        leading: const Icon(Icons.qr_code),
+        title: Text(
+          Loc.of(context).scanQrCode,
+          style: Theme.of(context).textTheme.bodyMedium,
+        ),
+        contentPadding: const EdgeInsets.symmetric(horizontal: Grid.side),
+        trailing: const Icon(Icons.chevron_right),
+        onTap: () => _scanQrCode(context, controller, errorText,
+            Loc.of(context).invalidDidQrCode, did),
+      ),
+    );
+  }
+
+  Future<bool> _isPhysicalDevice() async {
+    final deviceInfo = DeviceInfoPlugin();
+    return Platform.isIOS
+        ? (await deviceInfo.iosInfo).isPhysicalDevice
+        : Platform.isAndroid
+            ? (await deviceInfo.androidInfo).isPhysicalDevice
+            : false;
+  }
```

Comment:
I would move this out to a `provider` somewhere so we can use it again when needed.